int32 id
float32 velocity